### PR TITLE
CXX-2694 use setuptools rather than distutils (deprecated) for version calc

### PR DESCRIPTION
**Pull request by rcsanchez97**
Evergreen patch build: https://spruce.mongodb.com/version/646e3e3e9ccd4e1799389cce/tasks?sorts=STATUS%3AASC%3BBASE_STATUS%3ADESC

### CODE REVIEW

File: etc/calc_release_version.py [hunk 1]
```diff
@@ -190,7 +203,7 @@ def process_and_sort_tags(tags):
         tag_parts = tag.split('-')
         if len(tag_parts) >= 2 and tag_parts[0] not in processed_and_sorted_tags:
             processed_and_sorted_tags.append(tag)
-    processed_and_sorted_tags.sort(key=LooseVersion)
+    processed_and_sorted_tags.sort(key=Version)
```
Comment (kevinAlbs) [inline etc/calc_release_version.py:206]: Tasks appear to fail on this line:

https://spruce.mongodb.com/task/cxx_driver_debian10_release_latest_compile_and_test_with_static_libs_extra_alignment_patch_ed5a02fb0bf6135e25c45723ba5d768d59cb785e_646f9256d6d80a32b136916f_23_05_25_16_52_39/logs?execution=0

Comment (rcsanchez97) [reply etc/calc_release_version.py:206]: The failure seems to be a result of Evergreen executing the build while checked out to branch other than `master` (namely, the branch from the PR or `cxx-2694`). I'm working out a fix.

**Comment by rcsanchez97** [comment]
The latest commit seems to have resolved the issues. The tasks in the patch build appear to be succeeding now.

File: etc/calc_release_version.py [hunk 2]
```diff
@@ -196,16 +196,19 @@ def process_and_sort_tags(tags):
     for tag in raw_tags:
```
Comment (kevinAlbs) [inline etc/calc_release_version.py:196]: Update comment above to replace `LooseVersion comparison` with `Version comparison`

**Review by kevinAlbs — COMMENTED** [review]
Latest changes look good. I suggest waiting to merge until C driver tasks failures ([example](https://evergreen.mongodb.com/task/mongo_c_driver_windows_2017_debug_compile_nosasl_nossl_patch_6cbb08bd7cdd1dd7e7d30d374212a002955d0789_64776889d1fe07f4502985de_23_05_31_15_32_26)) related to https://github.com/mongodb/mongo-c-driver/commit/d9569499eaccd304f0f96766717729123c6b3428 are resolved (noted in slack here). That may suggest an additional change is needed.

File: etc/calc_release_version.py (hunk 2 above)
Comment (rcsanchez97) [reply etc/calc_release_version.py:201]: I pushed a change based on mongodb/mongo-c-driver@c35519cc5fb0

Comment (rcsanchez97) [reply etc/calc_release_version.py:201]: And another commit to fix the comment.

**Review by kevinAlbs — APPROVED** [review]

**Pull request merged**